pkg/executor/join/hash_join_probe_worker_v2.go

```diff
@@ -0,0 +1,187 @@
+// Copyright 2024 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package join
+
+import (
+	"sync/atomic"
+	"time"
+
+	"github.com/pingcap/failpoint"
+	"github.com/pingcap/tidb/pkg/util/chunk"
+)
+
+// ProbeWorkerV2 is the probe worker used in hash join v2
+type ProbeWorkerV2 struct {
+	probeWorkerBase
+	HashJoinCtx *HashJoinCtxV2
+	// We build individual joinProbe for each join worker when use chunk-based
+	// execution, to avoid the concurrency of joiner.chk and joiner.selected.
+	JoinProbe ProbeV2
+}
+
+func (w *ProbeWorkerV2) scanRowTableAfterProbeDone() error {
+	w.JoinProbe.InitForScanRowTable()
+	ok, joinResult := w.getNewJoinResult()
+	if !ok {
+		return nil
+	}
+
+	for !w.JoinProbe.IsScanRowTableDone() {
+		joinResult = w.JoinProbe.ScanRowTable(joinResult, &w.HashJoinCtx.SessCtx.GetSessionVars().SQLKiller)
+		if joinResult.err != nil {
+			w.HashJoinCtx.joinResultCh <- joinResult
+			return joinResult.err
+		}
+		if joinResult.chk.IsFull() {
+			err := triggerIntest(3)
+			if err != nil {
+				return err
+			}
+
+			w.HashJoinCtx.joinResultCh <- joinResult
+			ok, joinResult = w.getNewJoinResult()
+			if !ok {
+				return nil
+			}
+		}
+	}
+	if joinResult == nil {
+		return nil
+	} else if joinResult.err != nil || (joinResult.chk != nil && joinResult.chk.NumRows() > 0) {
+		w.HashJoinCtx.joinResultCh <- joinResult
+	} else if joinResult.chk != nil && joinResult.chk.NumRows() == 0 {
```

Comment:
when will this happen?

Reply:
> when will this happen?

This logic is from the origin hash join.  Maybe this happens when there are no rows be successfully joined.